### PR TITLE
JS: Map Reduce Filter documentation

### LINKED ISSUE #2847: [JS]: Map Reduce Filter
### Description

Directory

Topics/Map_Reduce_Filter/
The index will be added manually after merging.

Description
What are these functions?
Description of each function
Syntax of each function 
Example of each function 

### Domain

JavaScript

### Type of Contribution

Documentation

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by harshmathurx**
<hr>

## Description 📜

Map, Filter, and Reduce functions described  along with examples
Fixes #2847 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [ ] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [ ] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm a GWOC'21 contributor

<hr>

### CODE REVIEW

**Comment by harshmathurx** [comment]
@paramsiddharth please review

**Comment by paramsiddharth** [comment]
@Creatoon @dakshsethi @s4-2 Please review.

**Review by paramsiddharth — APPROVED** [review]

File: JavaScript/Topics/Map_Reduce_Filter/README.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+### Example
+
+```jsx
+let totalMarks = studentRecords.reduce( ((acc,emp) => acc+emp.marks), 0)console.log(totalMarks);// logs: 241
+```
+
+Here is a tweet by Steve Luscher to sum it up very well.
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Map/filter/reduce in a tweet:<br><br>map([🌽, 🐮, 🐔], cook)<br>=&gt; [🍿, 🍔, 🍳]<br><br>filter([🍿, 🍔, 🍳], isVegetarian)<br>=&gt; [🍿, 🍳]<br><br>reduce([🍿, 🍳], eat)<br>=&gt; 💩</p>&mdash; Steven Luscher (@steveluscher) <a href="https://twitter.com/steveluscher/status/741089564329054208?ref_src=twsrc%5Etfw">June 10, 2016</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
```
Comment (dakshsethi) [inline JavaScript/Topics/Map_Reduce_Filter/README.md:116]: Script tag never works in markdown
Instead you show a photo of that tweet.

Also add your name like this
### **Contributor:** [Harsh Mathur](https://github.com/harshmathurx)
`### **Contributor:** [Harsh Mathur](https://github.com/harshmathurx)`

**Review by dakshsethi — CHANGES_REQUESTED** [review]

File: JavaScript/Topics/Map_Reduce_Filter/README.md (hunk 1 above)
Comment (s4-2) [reply JavaScript/Topics/Map_Reduce_Filter/README.md:116]: @harshmathurx Please do the suggested correction which is provided by @dakshsethi .

Comment (harshmathurx) [reply JavaScript/Topics/Map_Reduce_Filter/README.md:116]: I shall, thank you

**Review by dakshsethi — APPROVED** [review]
Looks good to me!! 🔥🔥🔥

**Comment by harshmathurx** [comment]
@paramsiddharth could you please merge it?

**Pull request merged**